test/result_test.rb

```diff
@@ -62,6 +62,51 @@ def finish(result)
     end
   end
 
+  describe "#fetch" do
+    it "returns an options value" do
+      assert_equal "lee", @result.fetch("name")
+      assert_equal "lee", @result.fetch(:name)
+      assert_equal "lee", @result.fetch("--name")
+      assert_equal "bar", @result.fetch("long_option")
+      assert_equal "bar", @result.fetch(:long_option)
+      assert_equal "bar", @result.fetch("--long-option")
```

Comment:
I think this is a bit over-tested because it's not just testing `Result#fetch`, it's also testing `Result#get` and others. I think the test examples are good though, but all of them only need one assertion without the duplication

Reply:
You think it's ok to just leave the following two tests:

```ruby
assert_equal "lee", @result.fetch("name")
```

and

```ruby
e = assert_raises(KeyError) { @result.fetch("unexisting") }
assert_equal "key not found: 'unexisting'", e.message
```

I'm not pretty sure if remove `"returns the default value of an option when a value is not provided"` and `"returns nil when an option is not provided and it does not have a default value"`, because they test edge cases, especially the last one.

Reply:
> I'm not pretty sure if remove "returns the default value of an option when a value is not provided" and "returns nil when an option is not provided and it does not have a default value", because they test edge cases, especially the last one.

I think they should be kept too, yes, especially since they catch the `nil` issue I mentioned in another comment :)

Reply:
Cool, I just push those changes :-)